h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -42,27 +44,127 @@
+
+    private static class Work extends Iced<Work> {
+      private Algo algo;
+      private int count;
+      private JobType type;
+      private int share;
+
+      Work(Algo algo, int count, JobType type, int share) {
+        this.algo = algo;
+        this.count = count;
+        this.type = type;
+        this.share = share;
+      }
+
+      int consume() {
+        return consume(1);
+      }
+
+      int consume(int amount) {
+        int c = Math.min(this.count, amount);
+        this.count -= c;
+        return c * this.share;
+      }
+
+      int consumeAll() {
+        return consume(Integer.MAX_VALUE);
+      }
+    }
+
+    private boolean canAllocate = true;
+    private final LinkedList<Work> allocations = new LinkedList<>();
+
+    WorkAllocations allocate(Algo algo, int count, JobType type, int workShare) {
+      if (!canAllocate) throw new IllegalStateException("can't allocate new work");
+
+      allocations.add(new Work(algo, count, type, workShare));
+      return this;
+    }
+
+    void end() {
+      canAllocate = false;
+    }
+
+    void remove(Algo algo) {
+      final Iterator<Work> iter = allocations.iterator();
+      while (iter.hasNext()) {
+        if (algo.equals(iter.next().algo)) iter.remove();
+      }
+    }
+
+    Work getAllocation(Algo algo, JobType workType) {
+      for (Work alloc : allocations) {
+        if (alloc.algo == algo && alloc.type == workType) return alloc;
+      }
+      return null;
+    }
+
+    private int sum(List<Work> workItems) {
+      int tot = 0;
+      for (Work item : workItems) {
+        tot += (item.count * item.share);
+      }
+      return tot;
+    }
+
+    int remainingWork() {
+      return sum(allocations);
+    }
+
+    int remainingWork(JobType type) {
+      List<Work> selected = new ArrayList<>(allocations.size());
+      for (Work item : allocations) {
+        if (item.type == type) selected.add(item);
+      }
+      return sum(selected);
+    }
+
+  }
+
   private final static boolean verifyImmutability = true; // check that trainingFrame hasn't been messed with
   private final static SimpleDateFormat fullTimestampFormat = new SimpleDateFormat("yyyy.MM.dd HH:mm:ss.S");
   private final static SimpleDateFormat timestampFormatForKeys = new SimpleDateFormat("yyyyMMdd_HHmmss");
 
-  // TODO: UGH: this should be dynamic, and it's easy to make it so
-  //   just turn this enum into a class...
+  // if we need to make the Algo list dynamic, we should just turn this enum into a class...
   // NOTE: make sure that this is in sync with the exclude option in AutoMLBuildSpecV99
-  public enum algo {
+  public enum Algo implements algo {
     GLM,
     DRF,
     GBM,
     DeepLearning,
     StackedEnsemble,
-//    XGBoost,
+    XGBoost,
     ;
 
     String urlName() {
       return this.name().toLowerCase();
     }
   }
 
+  /**
+   * for external use only.
+   * Keeping this mapping for customers consuming the Java API directly
```

Comment:
Please use the term "users" instead of "customers". Can you also include a note about when will this compatibility interface be removed? Eg.: deprecated, will be removed in H2O 3.24.0.1, please use ...